clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/provider/agent/KubernetesControllersCachingAgent.groovy

```diff
@@ -0,0 +1,448 @@
+      def parse = Keys.parse(it)
+      if (parse && namespaces.contains(parse.namespace) && parse.account == accountName) {
+        return it
+      } else {
+        return null
+      }
+    }
+
+    def keyCount = keys.size()
+    def be = keyCount == 1 ? "is" : "are"
+    def pluralize = keyCount == 1 ? "" : "s"
+    log.info("There $be $keyCount pending on demand request$pluralize")
+
+    providerCache.getAll(Keys.Namespace.ON_DEMAND.ns, keys).collect {
+      [
+        details  : Keys.parse(it.id),
+        cacheTime: it.attributes.cacheTime,
+        processedCount: it.attributes.processedCount,
+        processedTime: it.attributes.processedTime
+      ]
+    }
+  }
+
+  @Override
+  String getSimpleName() {
+    return KubernetesControllersCachingAgent.getSimpleName()
+  }
+
+
+  V1beta1StatefulSet loadStatefulSet(String namespace, String name) {
+    credentials.apiClientAdaptor.getStatefulSet(namespace, name)
+  }
+  V1beta1StatefulSetList getStatefulSets(String namespace) {
+    return credentials.apiClientAdaptor.getStatefulSets(namespace)
+  }
+
+
+  /*V1beta1StatefulSet  getStatefulSet(String namespace, String serverGroupName) {
+    return this.appsV1beta1Api.readNamespacedStatefulSet(serverGroupName, namespace , null , false, false)
+  }*/
+  List<V1beta1StatefulSet> loadStatefulSets() {
+    namespaces.collect { String namespace ->
+      credentials.apiClientAdaptor.getStatefulSets(namespace)
+    }.flatten()
+  }
+
+
+  private CacheResult buildCacheResult(List<StateFulSet> serverGroups, Map<String, CacheData> onDemandKeep, List<String> onDemandEvict, Long start) {
+    log.info("Describing items in ${agentType}")
+
+    Map<String, MutableCacheData> cachedApplications = MutableCacheData.mutableCacheMap()
+    Map<String, MutableCacheData> cachedClusters = MutableCacheData.mutableCacheMap()
+    Map<String, MutableCacheData> cachedServerGroups = MutableCacheData.mutableCacheMap()
+    Map<String, MutableCacheData> cachedInstances = MutableCacheData.mutableCacheMap()
+    Map<String, MutableCacheData> cachedLoadBalancers = MutableCacheData.mutableCacheMap()
+
+    // Map namespace -> name -> event
+    Map<String, Map<String, Event>> rcEvents = [:].withDefault { _ -> [:] }
+    Map<String, Map<String, Event>> rsEvents = [:].withDefault { _ -> [:] }
+    try {
+      namespaces.each { String namespace ->
+        rcEvents[namespace] = credentials.apiAdaptor.getEvents(namespace, KubernetesUtil.DEPRECATED_SERVER_GROUP_KIND)
+        rsEvents[namespace] = credentials.apiAdaptor.getEvents(namespace, KubernetesUtil.SERVER_GROUP_KIND)
+      }
+    } catch (Exception e) {
+      log.warn "Failure fetching events for all server groups in $namespaces", e
+    }
+
+    // Map namespace -> name -> autoscaler
+    Map<String, Map<String, HorizontalPodAutoscaler>> rcAutoscalers = [:].withDefault { _ -> [:] }
+    Map<String, Map<String, HorizontalPodAutoscaler>> rsAutoscalers = [:].withDefault { _ -> [:] }
```

Comment:
These rs/rc prefixes refer to "replica set" and "replication controller" respectively. I don't think they make sense to use here.